### ISSUE TITLE
Migrate to Rust 2021 Edition

### ISSUE THREAD

**Issue by idavis** [qir-alliance/pyqir#123]
With the release of [Rust 1.56](https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html), the 2021 Rust Edition has been available. The there is a [migration guide](https://doc.rust-lang.org/edition-guide/editions/transitioning-an-existing-project-to-a-new-edition.html) to new edition.

**Comment by divshacker** [comment]
Hey, I am interested in working on this issue. 
I have done a try updating the new version of rust but facing some errors. 

- `cargo fix --edition` works but throws a error
``` Caused by:
  process didn't exit successfully: `/home/divshacker/pyqir/target/debug/build/llvm-ir-c75243ba21a4165f/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'llvm-ir: Please select an LLVM version using a Cargo feature.', /home/divshacker/.cargo/registry/src/github.com-1ecc6299db9ec823/llvm-ir-0.8.1/build.rs:22:14
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed ```
- `Cargo.toml` don't have the edition field. 
Please help me out by resolving this error. 

Thanks

**Comment by idavis** [comment]
Hi @divshacker, thank you for your interest.

Please take a look at the [development guide on how to build](https://github.com/qir-alliance/pyqir/blob/main/docs/development-guide/building.md#building-pyqir-from-source) PyQIR. The build script must be run prior to using cargo commands. Then you can use the commands in the [development](https://github.com/qir-alliance/pyqir/blob/main/docs/development-guide/building.md#development) section of the docs.

If you want to use an existing installation of LLVM, you can set the [environment variables](https://github.com/qir-alliance/pyqir/blob/main/qirlib/README.md#environment-variables) so that the build doesn't have to compile LLVM.

**Comment by divshacker** [comment]
I have tried to run `build.sh` but it is exiting every time and not building the LLVM completely. First, I have tested it in my system, then I used docker and created the environment and ran `build.sh` but still throws error that cargo can't LLVM automatically. I don't know what happens but still this the problem. I think person who have already a working enviornment can run `cargo fix --edition ` to fix this issue, I have tried to build a working environment but fail every time
